src/core/meshcop/dataset_local.cpp

```diff
@@ -193,5 +215,73 @@ Error DatasetLocal::Save(const Dataset &aDataset)
     return error;
 }
 
+#if OPENTHREAD_CONFIG_PLATFORM_KEY_REFERENCES_ENABLE
+void DatasetLocal::DestroyItsKeys() const
+{
+    namespace Its = Crypto::Storage;
+
+    Its::KeyRef networkKeyRef = IsActive() ? Its::kActiveDatasetNetworkKeyRef : Its::kPendingDatasetNetworkKeyRef;
+    Its::KeyRef pskcRef       = IsActive() ? Its::kActiveDatasetPskcRef : Its::kPendingDatasetPskcRef;
+
+    Its::DestroyKey(networkKeyRef);
+    Its::DestroyKey(pskcRef);
+}
+
+void DatasetLocal::StoreItsKeys(Dataset &aDataset) const
+{
+    namespace Its = Crypto::Storage;
+
+    Its::KeyRef    networkKeyRef = IsActive() ? Its::kActiveDatasetNetworkKeyRef : Its::kPendingDatasetNetworkKeyRef;
+    Its::KeyRef    pskcRef       = IsActive() ? Its::kActiveDatasetPskcRef : Its::kPendingDatasetPskcRef;
+    NetworkKeyTlv *networkKeyTlv = static_cast<NetworkKeyTlv *>(aDataset.GetTlv(Tlv::kNetworkKey));
+    PskcTlv *      pskcTlv       = static_cast<PskcTlv *>(aDataset.GetTlv(Tlv::kPskc));
+
+    if (networkKeyTlv != nullptr)
+    {
+        SuccessOrAssert(Its::ImportKey(networkKeyRef, Its::kKeyTypeRaw, Its::kKeyAlgorithmVendor, Its::kUsageExport,
+                                       Its::kTypePersistent, networkKeyTlv->GetNetworkKey().m8, NetworkKey::kSize));
+        NetworkKey networkKey;
```

Comment:
Maybe some comment here could be useful for someone to understand why we do next 3 lines - i mean, why we clear the network key.